datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -419,92 +423,154 @@ impl GroupedHashAggregateStream {
-                        .and(Ok(()))?;
-                    // normal accumulators
-                    group_state
-                        .accumulator_set
-                        .iter_mut()
-                        .zip(normal_values.iter())
-                        .map(|(accumulator, aggr_array)| {
-                            (
-                                accumulator,
-                                aggr_array
-                                    .iter()
-                                    .map(|array| {
-                                        // 2.3
-                                        array.slice(offsets[0], offsets[1] - offsets[0])
-                                    })
-                                    .collect::<Vec<ArrayRef>>(),
-                            )
-                        })
-                        .try_for_each(|(accumulator, values)| {
-                            let size_pre = accumulator.size();
-                            let res = match self.mode {
-                                AggregateMode::Partial => {
+                                })
+                                // 2.5
+                                .and(Ok(()))?;
+                            // normal accumulators
+                            group_state
+                                .accumulator_set
+                                .iter_mut()
+                                .zip(normal_values.iter())
+                                .map(|(accumulator, aggr_array)| {
+                                    (
+                                        accumulator,
+                                        aggr_array
+                                            .iter()
+                                            .map(|array| {
+                                                // 2.3
+                                                array.slice(
+                                                    offsets[0],
+                                                    offsets[1] - offsets[0],
+                                                )
+                                            })
+                                            .collect::<Vec<ArrayRef>>(),
+                                    )
+                                })
+                                .try_for_each(|(accumulator, values)| {
                                     accumulator.update_batch(&values)
-                                }
-                                AggregateMode::FinalPartitioned
-                                | AggregateMode::Final => {
+                                })
+                                // 2.5
+                                .and({
+                                    group_state.indices.clear();
+                                    Ok(())
+                                })
+                        })?;
+                }
+                AggregateMode::FinalPartitioned | AggregateMode::Final => {
```

Comment:
Maybe github is rendering the diff confusingly, but this seems like a significant amount of new code